hicommonwealth/commonwealth
# Unifying members count (Part 1/2)

## Link to Issue
Closes: #7877

## Description of Changes
- Adds schema and async script to populate profile_count of community table.
- Adds application level code in order to update profile_count when new profiles join a community.

## Test Plan
Migration script:
- After running db schema migration, run `pnpm run async-migrate-community-counts`.
- Check the Communities table. thread_count and profile_count should be filled out
- Tested on frack, works correctly.

Application code:
- Create a new address on a community with no user. Check that `select profile_count from "Communities" where id = :community_id;` increments by 1.
- Navigate to a separate community, join this community, check that the above query increments by 1 for this community
- Link a new address to an existing user, go to a 3rd community and join it, choosing the second address from the dropdown. Ensure that above query increments the profile_count by 1 for this 3rd community. 

## Deployment Plan
1. After deployment run the `pnpm run async-migrate-community-counts`

## Other Considerations
Followup ticket will be needed to be teed up to swap the UI to use the profile_count and remove old address_count.
https://github.com/hicommonwealth/commonwealth/issues/8079

Explain Analyze:
https://explain.dalibo.com/plan/fd85876d758b8853
